src/app/Controllers/ParameterValidator.cs

```diff
@@ -51,25 +54,88 @@ public static JsonResult Common(IQueryCollection query, string q, int pageNumber
             {
                 if (!int.TryParse(query["pageSize"], out int val) || val != pageSize || pageSize < 1 || pageSize > 1000)
                 {
-                    return GetAndLogBadParam("Invalid PageSize parameter", method, logger);
+                    return GetAndLogInvalidPageSizeParameter(method, logger);
                 }
             }
 
             return null;
         }
 
-        public static JsonResult GetAndLogBadParam(string message, string method, ILogger logger)
+        /// <summary>
+        /// validate search parameter
+        /// </summary>
+        /// <param name="method"></param>
+        /// <param name="logger"></param>
+        /// <returns></returns>
+        private static JsonResult GetAndLogInvalidSearchParameter(string method, ILogger logger)
         {
-            logger.LogWarning($"InvalidParameter|{method}|{message}");
+            const HttpStatusCode statusCode = HttpStatusCode.BadRequest;
+            const InnerErrorType innerErrorType = InnerErrorType.SearchParameter;
+            const string message = "Invalid q (search) parameter";
```

Comment:
rather than repeat these constants, I would encapsulate into one function - probably just pass an error type and have the function handle everything including figuring out the method

Reply:
Removed all constants in favor of passing types.